### PR TITLE
+ basic auth

### PR DESCRIPTION
**Pull request by pape-io**
Adds basic auth to the auth interceptor.

Thx to Simplificator for contributing it @nethad @nerdinand @cedricwider

### CODE REVIEW

File: lib/lhc-core-interceptors/auth.rb [hunk 1]
```diff
@@ -2,10 +2,20 @@ class LHC::Auth < LHC::Interceptor
...
+    authenticate!(request, options)
+  end
+
+  private
+
+  def authenticate!(request, options)
+    if options[:bearer]
+      token = options[:bearer]
       token = token.call if token.is_a?(Proc)
       request.headers['Authorization'] = "Bearer #{token}"
```
Comment (acoiro) [inline lib/lhc-core-interceptors/auth.rb]: You could set the `request.headers['Authorisation']` outside the if/else to avoid repeating it.

**Comment by acoiro** [comment]
super small commenter, rest is 👍

**Comment by pape-io** [comment]
@cedricwider what do you think?

**Comment by cedricwider** [comment]
👍 (apart from the fact that rubocop is unhappy, I approve this pull request)

**Pull request merged**